### DISCUSSION TITLE
Is there another good way to avoid checking for hidden elements at submitting?

### DISCUSSION THREAD

**Discussion by Connor-lipalater** [colinhacks/zod#2770] [Q&A] [answered]
When using useForm,I'd like to use a specific check rule based on the page element being shown and hidden, but I can't move forward with this error,or is there another good way to avoid checking for hidden elements at submitting,How to make `shouldUnregister` take effect in zod?

<img width="627" alt="image" src="https://github.com/colinhacks/zod/assets/105040671/d480330b-7320-4db5-9bf9-1219e94a2da2">
Error:There should be 3 parameters, but getting 1,types.d.ts(13, 74): no independent variable for "context" provided.
<img width="611" alt="image" src="https://github.com/colinhacks/zod/assets/105040671/8941bc0c-5df2-4596-8a7d-ce6ef923a96a">

**Comment by JacobWeisenburger** [comment, marked as answer]
Please send a full reproducible code example.
